app/models/system_setting.rb

```diff
@@ -1,2 +1,56 @@
 class SystemSetting < ApplicationRecord
+
```

Comment:
All the stuff around exchange type and comments below need to be thought through, and will also inform the state machine on Listing (which I started, but we need to more clearly define the workflow before I can move forward on that.)